src/Dazinator.Extensions.Options/Dazinator.Extensions.Options.csproj

```diff
@@ -1,11 +1,18 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>netstandard2.1</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Options" Version="2.2.0" />

```

Comment:
I'd like to keep support for netstandard2.0 which should probably actually be using the 2.1.0 dependency, and then make netstandard2.1 use the 3.1.2 dependency. Would you consider please have conditional package references here so we keep both target frameworks dependencies..

Reply:
@dazinator `v3.1.2` is also using supporting `netstandart 2.0`, or I'm missing something? Wouldn't `netstandart2.0` be enough?

![grafik](https://user-images.githubusercontent.com/17551597/77620501-cb6e2100-6f3a-11ea-9fe8-4adefac0b60a.png)
(https://www.nuget.org/packages/Microsoft.Extensions.Options/3.1.2)

Reply:
Yeah there are some different schools of thought here. Say you have a project targetting nestandard2.1. This means it must be running on a netstandard 2.1 compatible platform which at this time basically means a netcoreapp3.1 (LTS) application and therefore the compatible set of dependencies for a netcoreapp3.1 application are the 3.1 dependencies....  so by that logic I think it makes sense to set 3.1 deps as the baseline for netstandard 2.1.

It's interesting that the screenshot that you showed- ms supports netstandard 2.0 and netcoreapp3.1 but not netstandard2.1... I wondet why they dont swap netcoreapp3.1 for netstandard2.1 to gain wider support than just netcoreapp3.1. If they did they'd be online with what I am suggesting.

Yes - you can also install .netstandard2.0 dependencies into a netcoreapp3.1 app butn

Reply:
I see they also insist on 3.1 deps there even for netstandard2.0. Imho if you have a netstandard2.0 project it means it could be running in a  netcoreapp2.1 application (LTS) which means all your deps will be 2.1.x. If you installed their package into your project it would force some of your 2.1 deps to upgrade to 3.1 and this isnt supported in 2.1 applications - you can get breaking / incompatible changes amongst the binaries. So yeah.. their package looks awful to me - basically they dont care about existing 2.1 LTS applications they want to enforce upgrades to 3.1 and this means the package supporting netstandard2.0 is meaningless. I'm gonna create an issue over there with this rant.

Reply:
We had similar discussions and conversations in our company.

All dependencies on 2.0 are already included in 2.1. So if you don't need APIs from 2.1, it doesn't make sense for me to use netstandart 2.1. Internally, we have now changed all our libraries to netstandart 2.0 (previously we had them as multi target project; 2.0 and 2.1). Our server applications run as netcoreapp3.1 and all dependencies are netstandart 2.0.

The Microsoft documentation is not very helpful here...

Reply:
Yep I agree.
The problem I have seen with this approach in a few of my libraries is that when you compile a binary against an MS dependency - say Ms.Extensions.DependencyInjection version 2.1.0 and then you publish your library as a netstandard2.0 compatible package, and someone installs into their netcoreapp3.1 application, and they use aspnetcore 3.1 which means their  Ms.Extensions.DependencyInjection dependency > 3.1.0 - suddenly  my library breaks at runtime as their has been a binary incompatible change between the 2.1.0 version that my library was compiled against, and the 3.1.0 version of the library available at application runtime.

Reply:
By the same token I don't want my netstadnard2.0 library to force a consumer to upgrade all their dependencies to 3.1.x just in case their application is still on 2.1.0 (LTS)  - but that is what MS does if you want to use newer versions of thier packages as they are all versioned together.

Reply:
So maybe the safest way is to use [nuspec version ranges for dependencies](https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges)

> If NETSTANDARD2.0 use 2.1.X dependencies but don't allow upgrade to 3.0
   - deps are all: (2.1,3.0)

>  If NETSTANDARD2.0 use 3.1.X packages but don't allow upgrade to 4.0
   - deps are all: (3.1,4.0)

But I don't see anyone being this safe with their nuget packages at present. I'd like to find out from MS what their policy is re binary incompatibility changes - if it is only on major versions this can work, if its also on minor versions then this won't work.

Reply:
I've opened: https://github.com/dotnet/aspnetcore/issues/20183

Reply:
Thanks for your commitment. I will follow the progress of your issues, because it seems to be very interesting for me and at this point there is still a need for clarification! 👍